test/e2e/performanceprofile/functests/utils/profiles/profiles.go

```diff
@@ -100,27 +100,18 @@ func All() (*performancev2.PerformanceProfileList, error) {
 
 func UpdateWithRetry(profile *performancev2.PerformanceProfile) {
 	EventuallyWithOffset(1, func() error {
-		updatedProfile := &performancev2.PerformanceProfile{}
-		key := types.NamespacedName{
-			Name:      profile.Name,
-			Namespace: profile.Namespace,
-		}
-		// We should get the updated version of the performance profile.
-		// Otherwise, we will always try to update the profile with the old resource version
-		// and will always get the conflict error
-		if err := testclient.Client.Get(context.TODO(), key, updatedProfile); err != nil {
+		profileFromAPIServer := &performancev2.PerformanceProfile{}
+		// get the current resourceVersion
+		if err := testclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(profile), profileFromAPIServer); err != nil {
 			return err
 		}
-
-		updatedProfile.Spec = *profile.Spec.DeepCopy()
-		if err := testclient.Client.Update(context.TODO(), updatedProfile); err != nil {
+		profile = prepareForUpdate(profile, profileFromAPIServer)
```

Comment:
I'd still use DeepCopy() and reduce the variable aliasing:
```
prepared := prepareForUpdate(profile.DeepCopy(), profileFromAPIServer)
if err := testclient.Client.Update(context.TODO(), prepared); err != nil {
...
```

Reply:
Thanks. Done